Sources/PackageRegistry/SignatureValidation.swift

```diff
@@ -0,0 +1,257 @@
+            callbackQueue: callbackQueue
+        ) { result in
+            completion(
+                result.tryMap { signingEntity in
+                    // Always do signing entity TOFU check at the end,
+                    // whether the package is signed or not.
+                    self.signingEntityTOFU.validate(
+                        package: package,
+                        version: version,
+                        signingEntity: signingEntity,
+                        observabilityScope: observabilityScope,
+                        callbackQueue: callbackQueue,
+                        completion: completion
+                    )
+                }
+            )
+        }
+    }
+
+    private func getAndValidateSignature(
+        registry: Registry,
+        package: PackageIdentity.RegistryIdentity,
+        version: Version,
+        content: Data,
+        configuration: RegistryConfiguration.Security.Signing,
+        timeout: DispatchTimeInterval?,
+        observabilityScope: ObservabilityScope,
+        callbackQueue: DispatchQueue,
+        completion: @escaping (Result<SigningEntity?, Error>) -> Void
+    ) {
+        self.getSourceArchiveSignature(
+            registry: registry,
+            package: package,
+            version: version,
+            timeout: timeout,
+            observabilityScope: observabilityScope,
+            callbackQueue: callbackQueue
+        ) { result in
+            switch result {
+            case .success(let signature):
+                self.validateSignature(
+                    registry: registry,
+                    package: package,
+                    version: version,
+                    signature: signature.data,
+                    signatureFormat: signature.format,
+                    content: content,
+                    configuration: configuration,
+                    observabilityScope: observabilityScope,
+                    completion: completion
+                )
+            case .failure(let error):
+                guard let onUnsigned = configuration.onUnsigned else {
+                    return completion(.failure(
+                        RegistryError
+                            .missingConfiguration(details: "security.signing.onUnsigned")
+                    ))
+                }
+
+                switch onUnsigned {
+                case .error:
+                    completion(.failure(error))
+                case .prompt:
+                    if case RegistryError.sourceArchiveNotSigned = error {
+                        // Source archive is not signed
+                        // TODO: Call delegate to prompt user to continue with unsigned package or error.
+                    } else {
+                        // Cannot determine if source archive is signed or not
+                        // TODO: Call delegate to prompt user to continue with package as if it were unsigned or error.
+                    }
```

Comment:
maybe add fatalError or error callback calls here to make sure we dont forget to implement this